File: webapp/shared/lib/localeCurrency.ts
import { type CurrencyCode, defaultCurrency } from "./currency";

const localeCurrencyMap: Record<string, CurrencyCode> = {
  ja: "JPY",
  en: "USD",
  "zh-CN": "CNY",
  ko: "KRW",
  "zh-TW": "TWD",
  es: "EUR",
  "pt-BR": "USD",
};

export const getDefaultCurrencyForLocale = (
  locale: string | undefined
): CurrencyCode => localeCurrencyMap[locale ?? ""] ?? defaultCurrency;

File: mobileapp/lib/lib/locale_currency.dart
/// Locale-based default currency mapping.
///
/// Ported from webapp/shared/lib/localeCurrency.ts
library;

import 'currency.dart';

/// Mapping from locale to default currency
const localeCurrencyMap = <String, String>{
  'ja': 'JPY',
  'en': 'USD',
  'zh-CN': 'CNY',
  'ko': 'KRW',
  'zh-TW': 'TWD',
  'es': 'EUR',
  'pt-BR': 'USD',
};

/// Get the default currency for a given locale
String getDefaultCurrencyForLocale(String? locale) {
  if (locale == null) return defaultCurrency;
  return localeCurrencyMap[locale] ?? defaultCurrency;
}
